src/core/app.lua

```diff
@@ -283,18 +289,35 @@ function pace_breathing ()
    end
 end
 
+local jit_flush_if_needed = (function ()
+   local last_trigger = 0
+   return function (app)
+      local now = now()
+      if now < last_trigger then return end
+      if app.ingress_packet_drops then
+         local threshold = ingress_packet_drops.last_value + ingress_packet_drops.threshold
+         local current = app:ingress_packet_drops()
+         local exceeded = current > threshold
+         if exceeded then
+            ingress_packet_drops.last_value = current
+            last_trigger = now + ingress_packet_drops.wait
+            jit.flush()
+         end
+      end
+   end
+end)()
+
 function breathe ()
    monotonic_now = C.get_monotonic_time()
    -- Restart: restart dead apps
    restart_dead_apps()
    -- Inhale: pull work into the app network
    for i = 1, #app_array do
       local app = app_array[i]
---      if app.pull then
---         zone(app.zone) app:pull() zone()
       if app.pull and not app.dead then
          zone(app.zone)
          with_restart(app, app.pull)
+         jit_flush_if_needed(app)
```

Comment:
Finally we need to measure the overhead of this solution.  If we are processing 4e6 PPS and each breath pulls in 1e2 packets, that's 40000 breaths/second; probably more than needed.  I suggest 100ms as the sampling period.  WDYT?  We can then just add a timer to the engine and make the timer call `sample()`, which I guess could include the jit_flush_if_needed call; dunno.  Then we add a parameter to app.run() that controls whether we attach the timer or not.  WDYT?  I dunno, maybe they are bad ideas, I am just getting nervous about upstream's willingness to include such a thing in the core loop :)